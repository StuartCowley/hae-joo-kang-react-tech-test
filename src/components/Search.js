import React, { useState } from 'react';
import '../styles/search.css';

const Search = () => {
  return (
    <>
      <input className='search' type='text' />
      <button type='submit'>Search</button>
    </>
  );
}
 
export default Search