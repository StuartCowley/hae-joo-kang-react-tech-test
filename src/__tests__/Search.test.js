import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import Search from "../components/Search";

describe("Search", () => {
  const validProps = {
    mockSetSearchResults: jest.fn(),
    mockSetIsLoading: jest.fn()
  }

  it("renders Search correctly", () => {
    const rendered = renderer.create(
      <Search 
        setSearchResults={validProps.mockSetSearchResults}
        setIsLoading={validProps.mockSetIsLoading} 
      />
    );
    expect(rendered).toMatchSnapshot();
  });

  it("render a single button with correct text", () => {
    render(
      <Search 
        setSearchResults={validProps.mockSetSearchResults}
        setIsLoading={validProps.mockSetIsLoading} 
      />
    );

    const buttons = screen.getAllByRole("button");

    expect(buttons).toHaveLength(1);
    expect(buttons[0]).toHaveTextContent("Go!");
  });

  it("button calls correct function", async () => {
    render(
      <Search 
        setSearchResults={validProps.mockSetSearchResults}
        setIsLoading={validProps.mockSetIsLoading} 
      />
    );

    await fireEvent.click(screen.getByRole("button"));

    expect(validProps.mockSetSearchResults).toHaveBeenCalled();
    expect(validProps.mockSetIsLoading).toHaveBeenCalled();
  });
});
