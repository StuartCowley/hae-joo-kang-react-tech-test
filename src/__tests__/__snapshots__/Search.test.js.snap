// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search correctly 1`] = `
<div
  className="search"
>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <input
      className="search-input"
      onChange={[Function]}
      type="text"
    />
    <button
      className="search-button"
      type="submit"
    >
      Go!
    </button>
  </form>
</div>
`;
